lib/active_model/serializer.rb

```diff
@@ -46,6 +47,10 @@ def self.inherited(base)
       super
     end
 
+    def self.type(type)
+      self._type = type
+    end
```

Comment:
well, basically we want to avoid creating internal instance methods that collide with attribute names.. less important in class variables, but still :)